### PR TITLE
[linstor] Added a webhook linstorstorageclass and sds-drbd-operator templates

### PR DESCRIPTION
**Pull request by duckhawk**
## Description

Introduced a webhook for storage class validation which prevents manual creation and editing of storage classes with linstor.csi.linbit.com as the provisioner. Storage classes with this provisioner should now be created through the custom resource LinstorStorageClass. Additionally, Helm templates for deploying the new sds-drbd-operator have been added.

## Why do we need it, and what problem does it solve?

To ensure consistency and control over storage classes utilizing the linstor.csi.linbit.com provisioner, the new validation mechanism has been put in place. It guides users to utilize the LinstorStorageClass custom resource, thereby centralizing the management of such storage classes. 

## What is the expected result?

Upon applying these changes, any attempts to manually create or edit storage classes with the linstor.csi.linbit.com provisioner will be blocked by the webhook. Users will need to utilize the LinstorStorageClass custom resource for this purpose. The Helm templates for the sds-drbd-operator have been added as a preparatory step, setting the groundwork for its future deployment when the operator's code is finalized and integrated into Deckhouse.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: linstor
type: feature
summary: Added storage class validation webhook to restrict manual edits with linstor.csi.linbit.com provisioner and introduced Helm templates for the upcoming sds-drbd-operator.
```

### CODE REVIEW

File: modules/002-deckhouse/images/webhook-handler/werf.inc.yaml [hunk 1]
```diff
@@ -39,6 +45,11 @@ git:
   # - 'ee/fe/modules/*/webhooks/'
 - add: /{{ .ModulePath }}modules/002-deckhouse/images/webhook-handler/entrypoint.sh
   to: /entrypoint.sh
+- add: /{{ .ModulePath }}modules/002-{{ $.ModuleName }}/images/{{ $.ImageName }}/requirements.txt
+  to: /tmp/requirements.txt
```
[the remark this answers is not in the document]
Comment (duckhawk) [reply modules/002-deckhouse/images/webhook-handler/werf.inc.yaml:49]: It's done like in other packages

File: modules/041-linstor/templates/sds-drbd-operator/secret.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+---
+  {{ define "sds_drbd_data_nodes_node_selector" }}
+nodeSelector:
+  {{- . | toYaml | nindent 2 }}
+  {{- end }}
```
Comment (apolovov) [inline modules/041-linstor/templates/sds-drbd-operator/secret.yaml:5]:
```suggestion
{{ define "sds_drbd_data_nodes_node_selector" }}
nodeSelector:
{{- . | toYaml | nindent 2 }}
{{- end }}
```

File: modules/041-linstor/templates/sds-drbd-operator/secret.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  ---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: d8-sds-drbd-operator-config
+  namespace: d8-{{ .Chart.Name }}
+type: Opaque
+stringData:
+  config: |
+{{- if dig "dataNodes" "nodeSelector" false .Values.linstor }}
```
Comment (apolovov) [inline modules/041-linstor/templates/sds-drbd-operator/secret.yaml:15]: There should be default value for `linstor.dataNodes.nodeSelector` = `kubernetes.io/os: linux` in openapi.

File: modules/041-linstor/openapi/config-values.yaml [hunk 4]
```diff
@@ -6,6 +6,8 @@ properties:
     properties:
```
Comment (apolovov) [inline modules/041-linstor/openapi/config-values.yaml:6]:
```suggestion
    default: {}
    properties:
```

File: modules/041-linstor/template_tests/module_test.go [hunk 5]
```diff
@@ -43,6 +43,8 @@ const (
       master: 3
 `
 	moduleValues = `
+  dataNodes:
+    nodeSelector: "kubernetes.io/os: linux"
```
Comment (apolovov) [inline modules/041-linstor/template_tests/module_test.go:47]:
```suggestion
    nodeSelector:
      kubernetes.io/os: linux
```

File: modules/041-linstor/openapi/config-values.yaml [hunk 6]
```diff
@@ -3,9 +3,12 @@ properties:
...
     type: object
     description: Settings for Linstor on nodes with data
+    default: {}
     properties:
       nodeSelector:
         type: object
+        default:
+          kubernetes.io/os: linux
         additionalProperties:
           type: string
```
Comment (apolovov) [inline modules/041-linstor/openapi/config-values.yaml:13]:
```suggestion
        additionalProperties:
          type: object
          additionalProperties:
            type: string
```

File: modules/041-linstor/openapi/config-values.yaml [hunk 7]
```diff
@@ -3,9 +3,12 @@ properties:
   dataNodes:
     type: object
     description: Settings for Linstor on nodes with data
+    default: {}
     properties:
       nodeSelector:
         type: object
+        default:
+          kubernetes.io/os: linux
```
Comment (apolovov) [inline modules/041-linstor/openapi/config-values.yaml:11]:
```suggestion
          #kubernetes.io/os: linux
          aaa: bbb
```

File: modules/041-linstor/openapi/config-values.yaml [hunk 8]
```diff
@@ -3,9 +3,13 @@ properties:
   dataNodes:
     type: object
     description: Settings for Linstor on nodes with data
+    default: {}
```
Comment (apolovov) [inline modules/041-linstor/openapi/config-values.yaml:6]:
```suggestion
    #default: {}
```

File: modules/041-linstor/openapi/config-values.yaml [hunk 9]
```diff
@@ -3,9 +3,13 @@ properties:
   dataNodes:
     type: object
     description: Settings for Linstor on nodes with data
+    #default: {}
```
Comment (apolovov) [inline modules/041-linstor/openapi/config-values.yaml:6]:
```suggestion
    default: {}
```

File: modules/041-linstor/template_tests/module_test.go [hunk 10]
```diff
@@ -220,6 +223,7 @@ var _ = Describe("Module :: linstor :: helm template ::", func() {
 		})
 
 		It("Everything must render properly", func() {
+			//fmt.Println(f.KubernetesResource("Secret", "d8-linstor", "d8-sds-drbd-operator-config").ToYaml())
```
Comment (apolovov) [inline modules/041-linstor/template_tests/module_test.go:226]:
```suggestion
```

**Review by AleksZimin — APPROVED** [review]

**Review by name212 — APPROVED** [review]

**Review by apolovov — APPROVED** [review]

**Pull request merged**